modules/component-store/src/component-store.ts

```diff
@@ -111,4 +121,60 @@ export class ComponentStore<T extends object> {
       this.updater(stateOrUpdaterFn as (state: T) => T)();
     }
   }
+
+  /**
+   * Creates a selector.
+   *
+   * This supports chaining up to 4 selectors. More could be added as needed.
+   *
+   * @param projector A pure projection function that takes the current state and
+   *   returns some new slice/projection of that state.
+   * @return An observable of the projector results.
+   */
+  select<R>(projector: (s: T) => R): Observable<R>;
+  select<R, S1>(s1: Observable<S1>, projector: (s1: S1) => R): Observable<R>;
+  select<R, S1, S2>(
+    s1: Observable<S1>,
+    s2: Observable<S2>,
+    projector: (s1: S1, s2: S2) => R
+  ): Observable<R>;
+  select<R, S1, S2, S3>(
+    s1: Observable<S1>,
+    s2: Observable<S2>,
+    s3: Observable<S3>,
+    projector: (s1: S1, s2: S2, s3: S3) => R
+  ): Observable<R>;
+  select<R, S1, S2, S3, S4>(
+    s1: Observable<S1>,
+    s2: Observable<S2>,
+    s3: Observable<S3>,
+    s4: Observable<S4>,
+    projector: (s1: S1, s2: S2, s3: S3, s4: S4) => R
+  ): Observable<R>;
+  select<R>(...args: any[]): Observable<R> {
+    let observable$: Observable<R>;
+    const projector: (...args: any[]) => R = args.pop();
+    if (!args.length) {
```

Comment:
```suggestion
    if (args.length === 1) {
```

Reply:
It should actually be `args.lenght === 0`, since with `pop()` the projector function.
Adjusted to code and comments around it.

Reply:
Right, `pop` mutates the array.
It's better how it's now, I only based my comment based on the comment in the code.
Thanks Alex.